src/pgcopy/src/copy.rs

```diff
@@ -757,4 +847,70 @@ mod tests {
             Err("COPY delimiter and quote must be different".to_string())
         );
     }
+
+    #[mz_ore::test]
+    fn test_copy_csv_row() -> Result<(), io::Error> {
```

Comment:
Can these tests become a vec of some new TestCase struct? Or else datadriven? I think they could much more easily test things if they were tabularized or automated somehow.

Reply:
Agreed, would love to datadriven up this crate.

Reply:
Maybe better for a follow up though. I still want to get my full PR cleaned up and merged at some point.

Reply:
Yeah, that would be nice. In this case just generating the strings wasn't enough, because we'll also need the `RelationType` with all the column types etc. And the column type does matter in how the data is getting interpreted. Let me see if I can refactor. It's probably still going to look a bit clunky though.